pkg/adapterManager/manager.go

```diff
@@ -149,6 +159,21 @@ func (m *Manager) Execute(ctx context.Context, cfgs []*config.Combined, attrs at
 		}
 	}
 
+	bags := make([]*attribute.MutableBag, numCfgs)
```

Comment:
I agree that in the long run the conflict validation won't be useful, we can probably catch all this stuff up front at config load time rather than runtime. Until we get there though, I'd like to avoid surprises.

As for the issue of wasting an allocation, this array could easily be clumped into a per-request object pool as mentioned in the TODO statement a few lines up. I expect that once we start measuring end-to-end perf, we'll systematically identify and eliminate all the allocations in the main runtime paths. Anyway, I'll add a TODO here too.